meadow-client/src/MainFrame.purs

```diff
@@ -595,18 +605,23 @@ evalF (ResetSimulator next) = do
   resetContract
   pure next
 
---  mContents <- withMarloweEditor Editor.getValue
---  case mContents of
---    Nothing -> pure next
---    Just contents -> do
---      let contract = parse Parser.contract contents
---      case contract of
---        Right c -> do
---          assign _marloweCompileResult $ Left [MarloweError "oh no"]
---          pure next
---        Left err -> do
---          assign _marloweCompileResult $ Left [MarloweError err]
---          pure next
+evalF (Undo next) = do
+  modifying (_marloweState) removeState
+  mCurrContract <- use (_marloweState <<< _Head <<< _contract)
+  case mCurrContract of
+    Just currContract -> do
+      void $ withMarloweEditor $ Editor.setValue (show $ pretty currContract) (Just 1)
+      pure next
+    Nothing -> pure next
```

Comment:
Small thing, but I feel it's nicer to return the `next` in a single place, when you can:

```
  case mCurrContract of
    Just currContract ->
      void $ withMarloweEditor $ Editor.setValue (show $ pretty currContract) (Just 1)
    Nothing -> pure unit
  pure next
```

Reply:
ok, don't see the difference myself but happy to change